app/src/main/java/com/expidev/gcmapp/json/AssignmentsJsonParser.java

```diff
@@ -31,6 +31,12 @@
 
                 assignment.setId(assignmentJson.getString("id"));
                 assignment.setTeamRole(assignmentJson.getString("team_role"));
+                
+                JSONObject location = assignmentJson.getJSONObject("location");
```

Comment:
This will throw an exception when there is no location (when not a leader). I'll add a commit to fix this.